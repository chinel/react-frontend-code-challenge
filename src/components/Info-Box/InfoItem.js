import { Card } from "@material-ui/core";
import React from "react";
import InfoIcon from "../../assets/images/info-icon.svg";
import useStyles from "./InfoItemStyles";

function InfoItem() {
  const classes = useStyles();
  return (
    <Card className={classes.card}>
      <div className={classes.content}>
        <h3 className={classes.cardTitle}>2344</h3>
        <p className={classes.cardSubTitle}>Past metric</p>
      </div>
      <div>
        <img src={InfoIcon} alt="info" />
      </div>
    </Card>
  );
}

export default InfoItem;
