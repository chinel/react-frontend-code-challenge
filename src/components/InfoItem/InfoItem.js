import { Card } from "@material-ui/core";
import React from "react";
import InfoIcon from "../../assets/images/info-icon.svg";
import useStyles from "./InfoItemStyles";

function InfoItem({ title, total }) {
  const classes = useStyles();
  return (
    <Card className={classes.card}>
      <div className={classes.content}>
        <h3 className={classes.cardTitle}>{total} </h3>
        <p className={classes.cardSubTitle}>
          <span style={{ textTransform: "capitalize" }}>{title}</span> metric
        </p>
      </div>
      <div>
        <img src={InfoIcon} alt="info" />
      </div>
    </Card>
  );
}

export default InfoItem;
