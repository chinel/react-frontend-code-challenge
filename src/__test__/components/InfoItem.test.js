import React from "react";
import { render } from "@testing-library/react";
import useStyles from "../../components/InfoItem/InfoItemStyles";
import InfoItem from "../../components/InfoItem/InfoItem";

function checkStyles() {
  let values;

  function MockedComponent() {
    values = useStyles();
    return null;
  }
  render(<MockedComponent />);
  return values;
}

describe("<InfoItem/>", () => {
  it("renders the <InfoItem/> with populated data", () => {
    const { container, getByText } = render(
      <InfoItem title="ongoing" total={100} />
    );

    expect(getByText("ongoing")).toBeTruthy();
    expect(getByText("100")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("validate the return values of useStyles hook", () => {
    const validateStlyes = checkStyles();

    expect(validateStlyes.card).toBeTruthy();
  });
});
