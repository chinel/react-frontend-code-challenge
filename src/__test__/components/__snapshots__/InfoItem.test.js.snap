// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoItem/> renders the <InfoItem/> with populated data 1`] = `
<div
  class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="makeStyles-content-2"
  >
    <h3
      class="makeStyles-cardTitle-3"
    >
      100
       
    </h3>
    <p
      class="makeStyles-cardSubTitle-4"
    >
      <span
        style="text-transform: capitalize;"
      >
        ongoing
      </span>
       metric
    </p>
  </div>
  <div>
    <img
      alt="info"
      src="info-icon.svg"
    />
  </div>
</div>
`;
